test/boost/database_test.cc

```diff
@@ -1101,3 +1101,48 @@ SEASTAR_TEST_CASE(database_drop_column_family_clears_querier_cache) {
         BOOST_REQUIRE_EQUAL(qc.get_stats().population, 0);
     });
 }
+
+SEASTAR_TEST_CASE(drop_table_with_snapshots) {
+    sstring ks_name = "ks";
+    auto make_table_name = [gen = 0] () mutable {
+        return format("table{}", gen++);
+    };
+    auto make_timestamp_func = [] {
+        return [ts = db_clock::now()] { return make_ready_future<db_clock::time_point>(ts); };
+    };
+
+    // Allow auto_snapshot
+    // dir should exist
+    auto table_name = make_table_name();
+    co_await do_with_some_data({table_name}, [&] (cql_test_env& e) -> future<> {
+        auto cf_dir = e.local_db().find_column_family(ks_name, table_name).dir();
+        co_await e.local_db().drop_table_on_all(ks_name, table_name, make_timestamp_func(), true);
+        auto cf_dir_exists = co_await file_exists(cf_dir);
+        BOOST_REQUIRE_EQUAL(cf_dir_exists, true);
+        co_return;
+    });
+
+    // Disallow auto_snapshot
+    // dir should not exist
```

Comment:
I don't know what I'm missing, but it seems like you just have the same test three times, and it doesn't "allow" or "disallow" auto_snapshot anywhere?
I guess you'll go back to fix this in a later patch?

Reply:
`auto_snapshot` is allowed by default in the cql_test_env db::config.
The last boolean arg to `drop_table_on_all` is `with_snapshot`, that, when true, allows taking a snapshot, based on the `auto_snapshot` configuration, and when false, skips the snapshot.  We pass `true` the first time and `false` the second time.  As for the auto_snapshot config, this is done in a later patch.  Maybe I'll pull some of the logic here so it would make more sense.

BTW, I want to eventually change this boolean parameter to a tri-state class: `{auto, yes, no}` so its usage will be more visible, with the following semantics: auto: consider the auto_snapshot config, yes: take a snapshot, no: don't take a snapshot.  But it is out of scope for this series and will make it harder to backport.